scripts/run_formatters

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+
+# Runs clang-format and rustfmt on the entire codebase.
+# Use -a option to apply the formattings, otherwise the script just runs a check.
+# Exit codes if option -a is not present:
+#  - 1 there are files to be formatted.
+#  - 0 everything looks fine.
+# If -a is set, the return code is always 0.
+
+# Parse optionns (-h and -a).
+APPLY_FORMATTING=false
+while getopts "ha" opt; do
+  case "$opt" in
+  h|\?)
+    echo "Use -a option to apply the formatting."
+    exit 0
+    ;;
+  a)
+    APPLY_FORMATTING=true
+    ;;
+  esac
+done
+
+# Get CPU count.
+OS=$(uname)
+NPROC=1
+if [[ $OS = "Linux" ]] ; then
+  NPROC=$(nproc)
+elif [[ ${OS} = "Darwin" ]] ; then
+  NPROC=$(sysctl -n hw.physicalcpu)
+fi
+
+if [ "$APPLY_FORMATTING" = true ] ; then
```

Comment:
It seems that if APPLY_FORMATTING is set then this particular block is run. I think it would make sense to keep this out in a separate file, and assume that the author will run one or the other script (or both), rather than having to understand what flags to pass to this one to make it do what they want? i.e. I think we should have `format` and `check_format` scripts, as there is no overlap between them (apart from the multi-core stuff I guess).

Reply:
Split formatting and checking to 2 separate files.